### ISSUE TITLE
Result on VOT2018

### ISSUE THREAD

**Issue by xi-mao** [MegviiDetection/video_analyst#109]
I trained the network according to paper and got  AUC 56.2%  on LaSOT dataset. But the EAO on the VOT2018 dataset only 0.31. I don't know the reason. Is there something special I need to notice? The setting of the training env:
Pytorch1.1.0, CUDA10.0,
GPU：4*2080Ti
use fulldata.
Besides, the result of using the vot evaluation of this code is lower than using official code（i.e., matlab code） by 0.07. For example, when I test the VOT dataset using the model you offered, the evaluation of this code gives 0.35 EAO, but if I put the tested result into the official evaluation code, the result is 0.42 which is similar to the paper. Why?
Expect for your explanation, thanks~

**Comment by MARMOTatZJU** [comment]
@xi-mao Various factors contribute to this degradation of result as is mentioned [here](https://github.com/MegviiDetection/video_analyst/blob/master/docs/TUTORIALS/SOT_MODEL_ZOO.md#reproducibility).

The offered results was produced with some older version of packages (opencv / pytorch / numpy / etc.). VOT results are extremely sensitive to these due to its nature of restart mechanism. We consider it not appropriate for research and prefer to keep the dependencies' version free from the restriction needed by the reproduction of the result on this benchmark.

**Comment by xi-mao** [comment]
@MARMOTatZJU  Thanks for your explanation, VOT is indeed sensitive to hp. For the second question,  below is the tested results on VOT2018 by using the mode（i.e., siamfcpp-googlenet-vot-md5_f2680ba074213ee39d82fcb84533a1a6.pkl） this code offer                                                                                                    
. It shows 0.35 EAO. However, if I put the tested result into the official evaluation code, the result is 0.42 which is correct . Is there something working in the evaluation code?
--------------------------------------------------------------------                                                                                                         
|    Tracker Name    | Accuracy | Robustness | Lost Number |  EAO  |                                                                                                         
--------------------------------------------------------------------                                                                                                         
| siamfcpp_googlenet |  0.436   |   0.187    |    40.0     | 0.358 |                                                                                                         
-------------------------------------------------------------------- 
The tested env is under pytorch1.4, cuda10.1.

**Comment by lzx1413** [comment]
I tested the model with the latest master branch, the result is 

|    Tracker Name    | Accuracy | Robustness | Lost Number |  EAO  |
|-|-|-|-|-
| siamfcpp_googlenet |  0.581   |   0.173    |    37.0     | 0.427 |
You can try the latest code from master branch.

**Comment by MARMOTatZJU** [comment]
> The tested env is under pytorch1.4, cuda10.1.

There are several breaking changes for pytorch from 1.1.0 to 1.4.0, which are likely to contribute to this misalignement of performance.

> PyTorch version
e.g. Type Promotion between 1.2.0 and 1.3.0, see Type Promotion on [PyTorch release notes](https://github.com/pytorch/pytorch/releases).

Again, VOT is too sensitive in terms of benchmarking. Even the digit truncation of hyper-parameters (which you may see in our yaml config) can lead to a significant fluctuation of EAO.

**Comment by xi-mao** [comment]
@MARMOTatZJU  I also test under pytorch1.1, cuda10.0, it still shows 0.35EAO, but if I use the official evaluation code, it is 0.42.
The vot2018.json comes from [.](https://drive.google.com/drive/folders/10cfXjwQQBQeu48XMf2xc_W1LucpistPI).
The dataset is from [.](https://drive.google.com/drive/folders/18vaGhvrr_rt70sZr_TisrWl7meO9NE0J).  I'm confusing.

**Comment by xi-mao** [comment]
@lzx1413 
I try the lt branch, it still shows 0.35EAO, but if I use the official  [code](https://github.com/votchallenge/vot-toolkit) to evaluate the tested result,  it is 0.42. The vot2018.json comes from [here](https://drive.google.com/drive/folders/10cfXjwQQBQeu48XMf2xc_W1LucpistPI). The dataset is from [here](https://drive.google.com/drive/folders/18vaGhvrr_rt70sZr_TisrWl7meO9NE0J).  I'm confusing.

**Comment by lzx1413** [comment]
@xi-mao  can you provide  the version of other packages, such as numpy, numba, opencv

**Comment by xi-mao** [comment]
@lzx1413  I test on two version env.
-------------------------------
First：
python=3.6
pytorch=1.1.0
torchvision=0.3.0
cuda=10.0
numpy=1.18.5
numba=0.45.0
opencv=3.4.2
llvmlite=0.32.1
scipy=1.2.1
cpython=0.27.3
colorama=0.3.7
gcc version 7.5.0

--------------------------------
Second:
python=3.7
pytorch=1.4.0
torchvision=0.5.0
cuda=10.1.24
numpy=1.18.5
numba=0.45.0
opencv=4.2.0
llvmlite=0.32.1
scipy=1.2.1
cpython=0.27.3
colorama=0.3.7
ipython=7.16.1
gcc version 7.5.0
---------------------------------

**Comment by xi-mao** [comment]
@lzx1413 
Besides, cause the json file doesn't contain the key of 'width' and 'hight', I change this line a random digital（500，300). This should not influence accuracy.
 
https://github.com/MegviiDetection/video_analyst/blob/62db6ea739c9cee153a4c47473d72d69c3100b01/videoanalyst/evaluation/vot_benchmark/pysot/datasets/vot.py#L119

**Comment by lzx1413** [comment]
@xi-mao  we have offered one vot18 json file at https://github.com/MegviiDetection/video_analyst/blob/master/videoanalyst/evaluation/vot_benchmark/vot_list/vot2018/VOT2018.json, please use this one for test. It has the key of width and height

**Comment by xi-mao** [comment]
@lzx1413  Thanks, this solves my problem.
I have another question, do u search the best model or hyperparameters for VOT after training? I have train twice in different PyTorch version. Now the best result  EAO on VOT2018 is 0.37 without search model and hyperparameters.

**Comment by lzx1413** [comment]
@xi-mao  Yes, if you want to report one better result than other papers, searching the best model or hyperparameters for VOT is a necessary step.

**Comment by MARMOTatZJU** [comment]
@xi-mao Would you mind provide the package list which can reproduce the exact VOT result (with _pip freeze_ to print that)? We consider provide Docker image in the future to address the reproduction issue.

**Comment by xi-mao** [comment]
@MARMOTatZJU 
OK~
----------------torch1.1-------------------

absl-py==0.9.0
aiocontextvars==0.2.2
astroid==2.4.2
backcall==0.2.0
cachetools==4.1.0
certifi==2020.6.20
cffi==1.14.0
chardet==3.0.4
colorama==0.3.7
contextvars==2.4
cycler==0.10.0
Cython==0.27.3
decorator==4.4.2
future==0.18.2
google-auth==1.18.0
google-auth-oauthlib==0.4.1
grpcio==1.30.0
idna==2.9
imageio==2.8.0
immutables==0.14
importlib-metadata==1.7.0

numba==0.45.0
numpy==1.18.5
oauthlib==3.1.0
olefile==0.46
pandas==1.0.5
parso==0.7.0
pexpect==4.8.0
pickleshare==0.7.5
Pillow==7.1.2
prompt-toolkit==3.0.5
protobuf==3.12.2
ptyprocess==0.6.0
pyasn1==0.4.8
pyasn1-modules==0.2.8
pycocotools==2.0.1
pycparser==2.20
Pygments==2.6.1
pylint==2.5.3
pyparsing==2.4.7
python-dateutil==2.8.1
pytz==2020.1
PyWavelets==1.1.1
PyYAML==5.3.1
requests==2.24.0
requests-oauthlib==1.3.0
rsa==4.6
scikit-image==0.17.2
scipy==1.2.1
Shapely==1.7.0
six==1.15.0
tensorboard==2.2.2
tensorboard-plugin-wit==1.6.0.post3
tifffile==2020.6.3
toml==0.10.1
torch==1.1.0
torchvision==0.3.0
tqdm==4.46.1
traitlets==4.3.3
typed-ast==1.4.1
urllib3==1.25.9
wcwidth==0.2.5
Werkzeug==1.0.1
wget==3.2
wrapt==1.12.1
yacs==0.1.7
yapf==0.28.0
zipp==3.1.0

------------torch1.4--------------

absl-py==0.9.0
astroid==2.4.2
backcall==0.2.0
cachetools==4.1.1
certifi==2020.6.20
chardet==3.0.4
colorama==0.3.7
cycler==0.10.0
Cython==0.27.3
decorator==4.4.2
future==0.18.2
google-auth==1.18.0
google-auth-oauthlib==0.4.1
grpcio==1.30.0
idna==2.10
imageio==2.8.0
importlib-metadata==1.7.0
ipython==7.16.1
ipython-genutils==0.2.0
isort==4.3.21
jedi==0.17.1
kiwisolver==1.2.0
lazy-object-proxy==1.4.3
llvmlite==0.32.1
loguru==0.4.1
Markdown==3.2.2
matplotlib==3.2.2
mccabe==0.6.1
mkl-fft==1.1.0
mkl-random==1.1.1
mkl-service==2.3.0
networkx==2.4
numba==0.45.0
numpy==1.18.5
oauthlib==3.1.0
olefile==0.46
opencv-python==4.2.0.34
pandas==1.0.5
parso==0.7.0
pexpect==4.8.0
pickleshare==0.7.5
Pillow==7.1.2
prompt-toolkit==3.0.5
protobuf==3.12.2
ptyprocess==0.6.0
pyasn1==0.4.8
pyasn1-modules==0.2.8
pycocotools==2.0.1
Pygments==2.6.1
pylint==2.5.3
pyparsing==2.4.7
python-dateutil==2.8.1
pytz==2020.1
PyWavelets==1.1.1
PyYAML==5.3.1
requests==2.24.0
requests-oauthlib==1.3.0
rsa==4.6
scikit-image==0.17.2
scipy==1.2.1
Shapely==1.7.0
six==1.15.0
tensorboard==2.2.2
tensorboard-plugin-wit==1.7.0
tifffile==2020.6.3
toml==0.10.1
torch==1.4.0
torchvision==0.5.0
tqdm==4.47.0
traitlets==4.3.3
typed-ast==1.4.1
urllib3==1.25.9
wcwidth==0.2.5
Werkzeug==1.0.1
wget==3.2
wrapt==1.12.1
yacs==0.1.7
yapf==0.28.0
zipp==3.1.0

-------------------------------
